### PR TITLE
Minor neatening of some strategies

### PR DESCRIPTION
**Pull request by bootandy**
Consistency of strategy method: strategy method is an instance method not a static method in most implementations.

Darwin strategy: neaten use class variable

Strategies that use opponent.history should check opponent.history exists not self.history

### CODE REVIEW

File: axelrod/strategies/averagecopier.py [hunk 1]
```diff
@@ -19,8 +19,7 @@ class AverageCopier(Player):
         'manipulates_state': False
     }
 
-    @staticmethod
-    def strategy(opponent):
+    def strategy(self, opponent):
```
Comment (drvinceknight) [inline axelrod/strategies/averagecopier.py]: As was implemented here the strategy method was in fact static and not using self. By passing self you do make it a non static method but there is no advantage to doing so as the method does not use self.

**Comment by drvinceknight** [comment]
A part from 20f728c I am happy with this (making the methods static when possible to do so was one of the improvements made at some point to the library). I suggest reverting 20f728c.

**Comment by bootandy** [comment]
OK, static method commit reverted.

**Comment by drvinceknight** [comment]
:+1:

**Comment by marcharper** [comment]
Just FYI, we are working under the (unstated) assumption that len(self.history) == len(opponent.history), which allows more static methods. This looks good otherwise :+1:

**Comment by drvinceknight** [comment]
> Just FYI, we are working under the (unstated) assumption that len(self.history) == len(opponent.history), which allows more static methods. This looks good otherwise

Actually the changes made by 794f597 make it possible to change those strategies to static methods right? Using `opponent.history` allows for more static methods.

@bootandy could you add to those methods the @static decorator as by changing them to use the opponent's length they are now in effect static methods. What do you think @marcharper?

**Comment by drvinceknight** [comment]
Some other merges have caused a conflict (just raising in case this gets missed).

**Comment by bootandy** [comment]
Will resolve conflicts and submit new pull request with static decorators

**Comment by drvinceknight** [comment]
:+1:

**Pull request closed without merging**